lib/valkyrie/types.rb

```diff
@@ -97,12 +97,11 @@ def of(type)
         super.default([].freeze)
       end
 
-      def member(type)
-        warn "[DEPRECATION] .member has been renamed to .of in dry-types and this " \
-          "method will be removed in the next major version of Valkyrie. Please use " \
-          ".of instead. " \
-          "Called from #{Gem.location_of_caller.join(':')}"
-        super.default([].freeze)
+      # Override optional to provide a default because without it an
+      # instantiated Valkyrie::Resource's internal hash does not have values for
+      # every possible attribute, resulting in `MissingAttributeError`.
+      def optional
+        super.default(nil)
```

Comment:
This seems a little strange to me. Why would someone override no-arg `optional` when they can override `default`?

Maybe this should call `default(nil)` rather than `super.default(nil)` -- or be defined as something like `default(attr=nil)` to give a place for YARD, and just call `super`.

If the no-arg `optional` is most expressive for declarative purposes, it could be preserved with a comment referring to `default`.

Reply:
`#optional` and #default` do two different things. We can't override `default(nil)` because nobody's calling that, we can't call `default(nil)` because that won't mark the type as optional (which makes it a type that is allowed to be whatever it's type is OR nil)